plugin.video.srgssr_ch_replay/resources/lib/plugin.py

```diff
@@ -0,0 +1,537 @@
+            self.logger.debug("AkamaiHD video")
+            token_url = f"http://tp.srgssr.ch/akahd/token?acl={parsed_url.path}"
+            response = requests.get(token_url).json()
+            token = response["token"]["authparams"]
+            media_url += "?" + token
+        return media_url
+
+    def _set_inputstream_params(self, listitem, protocol, mime_type):
+        """If Inputstream Adaptive is enabled and available, configure it and update the ListItem"""
+        is_helper = inputstreamhelper.Helper(protocol)
+        if to_bool(self.settings.enable_inputstream_adaptive) and is_helper.check_inputstream():
+            listitem.setContentLookup(False)
+            listitem.setMimeType(mime_type)
+            listitem.setProperty("inputstream", is_helper.inputstream_addon)
+            listitem.setProperty("inputstream.adaptive.manifest_type", protocol)
+
+    def _add_subtitles(self, listitem, video_id):
+        """If subtitles are enable and available, add them to the ListItem"""
+        self.logger.debug(f"Getting subtitles for video {video_id}")
+        if to_bool(self.settings.enable_subtitles):
+            video_urn = f"urn:{self.bu}:episode:tv:{video_id}"
+            resp = self.subs_client.get_subtitles(video_urn)
+
+            subs = []
+            for asset in resp["data"]["assets"]:
+                if asset is not None:
+                    for sub in asset["hasSubtitling"]:
+                        subs.append(sub["identifier"])
+            if subs:
+                self.logger.debug(f"Found subtitles: {subs}")
+                listitem.setSubtitles(subs)
+
+    # ================================= Helper methods ==================================
+
+    def _add_video_to_directory(self, show: dict, episode: dict, media: dict):
+        """Helper that adds a "Video" item to the Directory"""
+        url_args = {
+            "video_id": episode.get("id"),
+            "media_id": media.get("id"),
+        }
+
+        vid_name = episode.get("title", "") + " - " + media.get("title", "")
+        vid_desc = media.get("description", "")
+        duration = int(media.get("duration", 0) // 1000)
+
+        self._add_item_to_directory(
+            vid_name,
+            self.router.url(mode="play_video", **url_args),
+            label2=vid_desc,
+            thumbnail_image=media.get("imageUrl", ""),
+            fanart=show.get("imageUrl", ""),
+            video_info={
+                "Title": vid_name,
+                "Duration": duration,
+                "Plot": vid_desc,
+                "Aired": episode.get("publishedDate", ""),
+            },
+            properties={"IsPlayable": "true"},
+        )
+
+    def _add_next_page_to_directory(
+        self,
+        current_page: int,
+        next_page_url: str,
+        number_of_episodes: int,
+        url_mode: str,
+        url_args: dict = None,
+    ):
+        """Helper method to adds a "next page" item to the directory"""
```

Comment:
Minor issue:
* The page starts always with 0 e.g. "Next page 0 of 4" and I guess it would make more sense to start with 1
* The total page gets sometimes lost after a while e.g "Next page 1 of 4" -> click -> "Next page 2 of ?" -> click -> "Next page 3 of ?"

Reply:
The next page always starts with 2 (as the inital load/page is always nr 1). The page counter seems to work now.